pyteal/ast/itxn_test.py

```
@@ -134,4 +135,268 @@ def test_InnerTxnBuilder_SetFields():
+                    pt.TxnField.fee: pt.Int(0),
+                }
+            ),
+        ),
+        None,
+    ),
+    # Error cases
+    (
+        pt.Int(1),
+        "add(pay,txn,appl)void",
+        [
+            {},
+            {TxnField.type_enum: TxnType.AssetTransfer},
+            {TxnField.type_enum: TxnType.ApplicationCall},
+        ],
+        None,
+        None,
+        pt.TealInputError,
+    ),
+    (
+        pt.Int(1),
+        "add(pay,txn,appl)void",
+        [
+            {TxnField.type_enum: pt.Int(10)},
+            {TxnField.type_enum: TxnType.AssetTransfer},
+            {TxnField.type_enum: TxnType.ApplicationCall},
+        ],
+        None,
+        None,
+        pt.TealTypeError,
+    ),
+    (
+        pt.Int(1),
+        "add(pay,txn,appl)void",
+        [
+            {TxnField.type_enum: TxnType.ApplicationCall},
+            {TxnField.type_enum: TxnType.AssetTransfer},
+            {TxnField.type_enum: TxnType.ApplicationCall},
+        ],
+        None,
+        None,
+        pt.TealInputError,
+    ),
+    (
+        pt.Int(1),
+        "add(application,account,asset)void",
+        [
+            pt.abi.Asset(),
+            pt.abi.Account(),
+            pt.abi.Asset(),
+        ],
+        None,
+        None,
+        pt.TealTypeError,
+    ),
+    (
+        pt.Int(1),
+        "add(application)void",
+        [
+            pt.Bytes(""),
+        ],
+        None,
+        None,
+        pt.TealTypeError,
+    ),
+    (
+        pt.Int(1),
+        "add(asset)void",
+        [
+            pt.Bytes(""),
+        ],
+        None,
+        None,
+        pt.TealTypeError,
+    ),
+    (
+        pt.Int(1),
+        "add(account)void",
+        [
+            pt.Int(1),
+        ],
+        None,
+        None,
+        pt.TealTypeError,
+    ),
+    (
+        pt.Int(1),
+        "add(uint64,uint64)void",
+        [pt.abi.String(), pt.abi.Uint64()],
+        None,
+        None,
+        pt.TealTypeError,
+    ),
```

I just thought of 2 more invalid cases to test: too few or too many arguments being passed. Based on the implementation I think too many args is silently accepted, while too few args will error but the message might not be the cleanest, since it'll be a generic array out of bounds error.